documentation/src/docs/streaming.md

```diff
@@ -0,0 +1,204 @@
+|`withSerializationSchema`|The serialization schema which describes how to turn events into byte messages.|
+|`withEventRouter`|The router function which determines the Routing Key for a given event.|
+|`enableMetrics`|true or false to enable/disable reporting Pravega metrics. Metrics is enabled by default.|
+
+### Parallelism
+`FlinkPravegaWriter` supports parallelization. Use the `setParallelism` method to configure the number of parallel instances to execute.
+
+### Event Routing
+Every event written to a Pravega Stream has an associated Routing Key.  The Routing Key is the basis for event ordering.  See the [Pravega Concepts](http://pravega.io/docs/latest/pravega-concepts/#events) for details.
+
+When constructing the `FlinkPravegaWriter`, please provide an implementation of `io.pravega.connectors.flink.PravegaEventRouter` which will guarantee the event ordering. In Pravega, events are guaranteed to be ordered at the segment level.
+
+For example, to guarantee write order specific to sensor id, you could provide a router implementation like below.
+```
+private static class SensorEventRouter<SensorEvent> implements PravegaEventRouter<SensorEvent> {
+        @Override
+        public String getRoutingKey(SensorEvent event) {
+            return event.getId();
+        }
+    }
+   
+```
+
+### Event Time Ordering
+
+For programs that use Flink's event time semantics, the connector library supports writing events in event time order. In combination with a Routing Key, this establishes a well-understood ordering for each key in the output stream.
+
+Use the method `FlinkPravegaUtils::writeToPravegaInEventTimeOrder` to write a given `DataStream` to a Pravega Stream such that events are automatically ordered by event time (on a per-key basis). Refer [here](https://github.com/pravega/flink-connectors/blob/7971206038b51b3cf0e317e194c552c4646e5c20/src/test/java/io/pravega/connectors/flink/FlinkPravegaWriterITCase.java#L93) for sample code.
+
+### Writer Modes
+Writer modes relate to guarantees about the persistence of events emitted by the sink to a Pravega Stream.  The writer supports three writer modes:
+1. **Best-effort** - Any write failures will be ignored hence there could be data loss.
+2. **At-least-once** - All events are persisted in Pravega. Duplicate events
+are possible, due to retries or in case of failure and subsequent recovery.
+3. **Exactly-once** - All events are persisted in Pravega using a transactional approach integrated with the Flink checkpointing feature.
+
+By default, the _At-least-once_ option is enabled and use `.withWriterMode(...)` option to override the value.
+
+See the [Pravega documentation](http://pravega.io/docs/latest/pravega-concepts/#transactions) for details on transactional behavior.
+
+# Metrics
+Metrics are reported by default unless it is explicitly disabled using `enableMetrics(false)` option.
+See [Metrics](metrics.md) page for more details on type of metrics that are reported._

```

Comment:
There is a spurious `_` at the end of the sentence.

Reply:
Fixed